### ISSUE TITLE
comment_references: false positive with markdown link syntax

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/linter#750]
the `rfc` link below shows us as broken.

SDK 1.25.0-dev.7.0

```dart
/// An implementation of [keyed-hash method authentication codes][rfc].
///
/// [rfc]: https://tools.ietf.org/html/rfc2104
///
/// HMAC allows messages to be cryptographically authenticated using any
/// iterated cryptographic hash function.
class Hmac extends Converter<List<int>, Digest> {
...
}
```

**Comment by pq** [comment]
It looks like this is an issue that percolates up from the analyzer (its parser is treating `[rfc]` as a `CommentReference` node where it shouldn't).  Fixing this properly would involve a repair there.

@kevmoo : can you provide context?  Is this happening frequently?  How common are links in dartdocs?

**Comment by kevmoo** [comment]
See https://travis-ci.org/dart-lang/markdown/jobs/323208414#L528

Three pretty valid examples in this file...

https://github.com/dart-lang/markdown/blob/0eb8ae2a23295666149f3df0b2b184869c189b82/lib/src/extension_set.dart#L16

@pq – could you open a bug in analyzer on this?

**Comment by pq** [comment]
@bwilkerson, @scheglov : should I open a fresh bug or just move this one to the SDK repo?

**Comment by bwilkerson** [comment]
I think this is purely an analyzer issue, so moving it makes sense to me.

**Comment by pq** [comment]
This issue was moved to dart-lang/sdk#31761
